spi/transfer-spi/src/test/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/DataRequestTest.java

```diff
@@ -32,4 +35,80 @@ void verifyNoDestination() {
         assertThrows(IllegalArgumentException.class, () -> DataRequest.Builder.newInstance().id(id).assetId(asset.getId()).build());
     }
 
+    @Test
+    void verifyCopy() {
+        DataRequest dataRequest = newSampleDataRequest();
+
+        var copy = dataRequest.copy();
+
+        assertSampleDataAddress(dataRequest, copy);
+    }
+
+    @Test
+    void verifyDeepCopy() {
+        DataRequest dataRequest = newSampleDataRequest();
+
+        var copy = dataRequest.copy();
+
+        var copyProperties = copy.getProperties();
+        copyProperties.put("foo", "new value");
+
+        assertEquals("bar", dataRequest.getProperties().get("foo"));
+    }
+
+    @Test
+    void verifyToBuilder() {
+        DataRequest dataRequest = newSampleDataRequest();
+
+        var copy = dataRequest.toBuilder().build();
+
+        assertSampleDataAddress(dataRequest, copy);
+    }
+
+    private DataRequest newSampleDataRequest() {
+        var properties = new HashMap<String, String>(1);
+        properties.put("foo", "bar");
```

Comment:
var key = FAKER.lorem().word();
var value = FAKER.lorem().word();
var properties = Map.of(key, value);